lib/kernel.ts

```diff
@@ -455,6 +474,147 @@ export class Kernel {
 				return this.replaceContract(context, this.adminSession()!, contract);
 			}),
 		);
+
+		await this.setUpRelationshipsStream(logContext, context);
+
+		const loadedRelationships = await this.query<RelationshipContract>(
+			logContext,
+			this.adminSession()!,
+			SCHEMA_RELATIONSHIPS,
+		);
+
+		context.debug('Loading relationships', {
+			relationships: loadedRelationships.length,
+		});
+		this.setRelationships(context, loadedRelationships);
+	}
+
+	/**
+	 * Setup relationships cache. Logic from jellyfish-worker
+	 * Relationship contracts are streamed from the DB, so this has the most up to date version of them.
+	 *
+	 * @param logContext
+	 * @param context
+	 */
+	private async setUpRelationshipsStream(
+		logContext: LogContext,
+		context: Context,
+	) {
+		this.relationshipsStream = await this.stream(
+			logContext,
+			this.adminSession()!,
+			{
+				anyOf: [SCHEMA_RELATIONSHIPS],
+			},
+		);
+
+		this.relationshipsStream.on('data', (change: StreamChange) => {
+			context.info('relationshipsStream data event', change);
+			const contract = change.after;
+			const contractType = change.contractType.split('@')[0];
+			if (contractType !== 'relationship') {
```

Comment:
Is this check necessary? If we get a non-relationship contract here then the stream implementation is really broken!

Reply:
Thanks for pointing this out. I hadn't noticed, this is great as I can make the typecast in the next line in a safe way.
fixed; was also a leftover from jellyfish-worker who streams different contract types.